tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckPlaceHolders.java

```diff
@@ -63,253 +80,406 @@ public CheckCLDR handleCheck(String path, String fullPath, String value, Options
-
-                        if (!skip) {
-                            result.add(new CheckStatus().setCause(this)
+                    }
+                    if (usageIsMonogram) {
+                        result.add(new CheckStatus().setCause(checkAccessor)
+                            .setMainType(CheckStatus.warningType)
+                            .setSubtype(Subtype.namePlaceholderProblem)
+                            .setMessage("“.” is discouraged when usage=monogram, in " + namePattern));
+                    }
+                }
+            } else {
+                lastModifiedField = modifiedField;
+                Set<Modifier> modifiers = modifiedField.getModifiers();
+                Field field = modifiedField.getField();
+                switch (field) {
+                case prefix: case suffix:
+                    if (usageIsMonogram) {
+                        result.add(new CheckStatus().setCause(checkAccessor)
+                            .setMainType(CheckStatus.errorType)
+                            .setSubtype(Subtype.invalidPlaceHolder)
+                            .setMessage("Disallowed when usage=monogram: {" + field + "…}"));
+                    }
+                    break;
+                default:
+                    final boolean monogramModifier = modifiers.contains(Modifier.monogram);
+                    final boolean allCapsModifier = modifiers.contains(Modifier.allCaps);
+                    if (!usageIsMonogram) {
+                        if (monogramModifier) {
+                            result.add(new CheckStatus().setCause(checkAccessor)
+                                .setMainType(CheckStatus.warningType)
+                                .setSubtype(Subtype.invalidPlaceHolder)
+                                .setMessage("-monogram is strongly discouraged when usage≠monogram, in {" + modifiedField + "}"));
+                        }
+                    } else if (usageIsMonogram) {
+                        if (!monogramModifier) {
+                            result.add(new CheckStatus().setCause(checkAccessor)
                                 .setMainType(CheckStatus.errorType)
                                 .setSubtype(Subtype.invalidPlaceHolder)
-                                .setMessage("Duplicate fields: " + entry));
+                                .setMessage("-monogram is required when usage=monogram, in {" + modifiedField + "}"));
+                        } else if (!allCapsModifier) {
+                            result.add(new CheckStatus().setCause(checkAccessor)
+                                .setMainType(CheckStatus.warningType)
+                                .setSubtype(Subtype.invalidPlaceHolder)
+                                .setMessage("-allCaps is strongly encouraged with -monogram, in {" + modifiedField + "}"));
                         }
                     }
+                }
+            }
+            lastModifiedField = modifiedField;
+        }
 
```

Comment:
this method is too long and too deeply indented -- should be divided into subroutines, else bug-prone and hard to understand/maintain